docs/developer/make_verify.md

```diff
@@ -0,0 +1,235 @@
+in kubeedge project.  
+It runs a script to update the vendor dependencies and then checks if there are 
+any uncommmited changes. 
+
+### Verify vendor and update vendor steps
+**3.1**. Setting shell options and Defining the Root Directory
+
+```shell
+set -o errexit
+set -o nounset
+set -o pipefail
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+```
+This set the **KUBEEDGE_ROOT** to the absolute path of the root directory of the project.
+
+**3.2**. Exectueing `hack/update-vendor.sh` to update the vendor dependencies.
+
+```shell
+${KUBEEDGE_ROOT}/hack/update-vendor.sh
+```
+and
+
+```shell
+for repo in $(kubeedge::util::list_staging_repos); do
+    go mod tidy    #clean up  the go.mode file
+    go mod vendor  #create a vendor directory with all dependencies.
+done
+```
+
+It will update go.mod and go.sum for each repository in the staging repos.  
+And then it will re-create a symlink in the vendor directoyr pointing to the corresponding staging respository
+
+```shell
+for repo in $(kubeedge::util::list_staging_repos); do
+  rm -fr "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
+  ln -s "../../../staging/src/github.com/kubeedge/${repo}/" "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
+done
+```
+
+From the above operations, the **`verify-vendor.sh`** and **`update-vendor.sh`** will help other packages and tools updated,
+and maintain the consistency and correctness in the project's dependencies.
+
+**3.3**. Check Status
+
+After the **`update-vendor.sh`** executed, **`check_status`** will check git status.  
+    &emsp;If result is 0, it means there are no uncommitted changes, and vendor updated successfully.  
+    &emsp;If result is not 0, it means there are still uncommitted changes, and vendor verify failed.
+
+```shell
+ret=$(kubeedge::git::check_status)
+```
+
+ 
+## 4.Verify codegen
+
+The sub-target **`verify-codegen`** will run its corrsponding shell script
+
+```shell
+verify-codegen:
+    hack\verify-codegen.sh
+```
+
+### Verify codegen steps
+
+**4.1**. Defining Variables and clean up environment
+
+```shell
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+DIFFROOT="${KUBEEDGE_ROOT}/pkg"
+TMP_DIFFROOT="${KUBEEDGE_ROOT}/_tmp/pkg"
+_tmp="${KUBEEDGE_ROOT}/_tmp"
+cleanup
+```
+
+**4.2**. Run code generation script and comparing directory
+
+```shell
+"${KUBEEDGE_ROOT}/hack/update-codegen.sh"
+diff -Naupr "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+```
+
+**4.3**. Updating original directory and checking difference
+
+```shell
+cp -a "${TMP_DIFFROOT}"/* "${DIFFROOT}"
+```
+
+This verification ensures that the code in the pkg directory is up-to-date with the latest code generation.  
+It does this by:    
+    &emsp;copying the current code to a temporary directory,  
+    &emsp;running the code generation script, and  
+    &emsp;then comparing the original code with the freshly generated code.   
```

Comment:
Is there something wrong with the format here?

![image](https://github.com/user-attachments/assets/afd1927b-b866-4b24-8157-35b5859fb2e8)

Reply:
Format correct, I added action sequence number here.